src/sinks/util/unix.rs

```diff
@@ -115,27 +112,28 @@ impl UnixSink {
                             error,
                             path: &self.path
                         });
-                        UnixSinkState::Backoff(self.next_delay())
+                        let delay = self.next_delay();
+                        let delay = Box::new(async move { Ok(delay.await) }.boxed().compat());
+                        UnixSinkState::Backoff(delay)
                     }
                     Ok(Async::Ready(stream)) => {
                         emit!(UnixSocketConnectionEstablished { path: &self.path });
                         self.backoff = Self::fresh_backoff();
                         let out = FramedWrite::new(stream, BytesCodec::new());
-                        UnixSinkState::Open(out)
+                        UnixSinkState::Open(CompatSink::new(out))
                     }
                 },
                 UnixSinkState::Backoff(ref mut delay) => match delay.poll() {
                     Ok(Async::NotReady) => return Ok(Async::NotReady),
-                    // Err can only occur if the tokio runtime has been shutdown or if more than 2^63 timers have been created
-                    Err(err) => unreachable!(err),
+                    Err(_) => unreachable!(),
```

Comment:
Did the type here change? If it's unit, I prefer to show that with `Err(()) => `.

Reply:
:+1: done in https://github.com/timberio/vector/pull/3095/commits/9c5708450da833faa66ce260b7e786c4b2e648dd